src/pyproject_hooks/_impl.py

```diff
@@ -169,8 +198,11 @@ def get_requires_for_build_wheel(self, config_settings=None):
         )
 
     def prepare_metadata_for_build_wheel(
-        self, metadata_directory, config_settings=None, _allow_fallback=True
-    ):
+        self,
+        metadata_directory: str,
+        config_settings: Optional[Dict[str, Any]] = None,
+        _allow_fallback: bool = True,
+    ) -> Optional[str]:
```

Comment:
The docstring's `rtype` is a `str`, should be updated if it can also return `None`.  IIUC, this will only return `None` if `_allow_fallback` is `True`, so an overload might be more appropriate.